freetextresponse/mixins.py

```diff
@@ -0,0 +1,34 @@
+"""
+Mixins for the Free Text Response XBlock
+"""
+import datetime
+import pytz
+from django.conf import settings
+from xblock.fields import DateTime
+from xblock.fields import Scope
+
+
+class EnforceDueDates(object):
+    """
+    xBlock Mixin to allow xblocks to check the due date
+    (taking the graceperiod into account) of the
+    subsection in which they are placed
```

Comment:
Xblocks *could* set further due dates but different logic would have to be written in the `_can_submit()`